src/components/SignUp/index.tsx

```diff
@@ -10,100 +9,140 @@ import {
-        setError(error.message);
-        return;
+      if (error instanceof FirebaseError) {
+        setError("root", {
+          message: getFirebaseAuthErrorMessage(error.code),
+        });
       }
     }
   };
 
-  const getLocation = async (): Promise<
-    undefined | { latitude: number; longitude: number }
-  > => {
-    const userLocation = {
-      latitude: 0,
-      longitude: 0,
-    };
-
-    // request permission for foreground location
-    const { status } = await requestForegroundPermissionsAsync();
-
-    // if permission is not granted, set an error and return
-    if (status !== "granted") {
-      return;
-    }
+  return (
+    <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
+      <View style={styles.container}>
+        <Text style={{ marginBottom: 15 }}>Sign up for SolSync</Text>
 
-    // if permission is granted, get the current location
-    const location = await getLastKnownPositionAsync({});
+        <Controller
+          control={control}
+          name="displayName"
+          render={({ field: { onChange, onBlur, value } }) => (
+            <TextInput
+              autoCorrect={false}
+              style={styles.input}
+              placeholder="Display Name"
+              onChangeText={onChange}
+              onBlur={onBlur}
+              value={value}
+            />
+          )}
+        />
 
-    userLocation.latitude = location?.coords.latitude || 0;
-    userLocation.longitude = location?.coords.longitude || 0;
+        {errors.displayName && (
+          <Text style={{ color: "red" }}>{errors.displayName.message}</Text>
+        )}
 
-    return userLocation;
-  };
+        <Controller
+          control={control}
+          name="email"
+          render={({ field: { onChange, onBlur, value } }) => (
+            <TextInput
+              autoCorrect={false}
+              style={styles.input}
+              placeholder="Email"
+              onChangeText={onChange}
+              onBlur={onBlur}
+              value={value}
+            />
+          )}
+        />
+        {errors.email && (
+          <Text style={{ color: "red" }}>{errors.email.message}</Text>
+        )}
 
-  return (
-    <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
-      <View
-        style={{
-          flex: 1,
-          backgroundColor: "#fff",
-          justifyContent: "center",
-          alignItems: "center",
-        }}
-      >
-        <Text style={{ marginBottom: 15 }}>Sign up for SolSync</Text>
-        <TextInput
-          autoCorrect={false}
-          style={styles.input}
-          placeholder="Email"
-          onChangeText={setEmail}
-          value={email}
+        <Controller
+          control={control}
+          name="password"
+          render={({ field: { onChange, onBlur, value } }) => (
+            <TextInput
+              autoCorrect={false}
+              style={styles.input}
+              placeholder="Password"
```

Comment:
Make password hidden while you type it in

```suggestion
              placeholder="Password"
              secureTextEntry={true}
```